src/components/basket-item/basket-item.jsx

```diff
@@ -4,27 +4,69 @@ var React = require('react');
 module.exports = React.createClass({
   propTypes: {
     currencySymbol: React.PropTypes.string,
+    freeText: React.PropTypes.string,
+    removeText: React.PropTypes.string,
     price: React.PropTypes.number,
+    title: React.PropTypes.node,
+    toggleDescription: React.PropTypes.bool,
+    handleRemove: React.PropTypes.func,
     children: React.PropTypes.node
   },
+  getInitialState: function() {
+    return {
+      descriptionVisibility: !this.props.toggleDescription
+    };
+  },
   getDefaultProps: function() {
     return {
-      currencySymbol: '£'
+      title: null,
+      price: null,
+      handleRemove: null,
+      currencySymbol: '£',
+      freeText: 'FREE',
+      removeText: 'remove',
+      toggleDescription: false
     };
   },
-  render: function() {
-    var priceNode;
-    if (this.props.price) {
-      priceNode = (
-        <div className="component-basket-item-price">
-          <span className="component-currency">{this.props.currencySymbol}</span>{this.props.price}
-        </div>
-      );
+  toggleDescriptionVisibility: function() {
+    this.setState({ descriptionVisibility: !this.state.descriptionVisibility });
+  },
+  titleNode: function() {
+    if (this.props.title === null) return null;
+    if (this.props.toggleDescription) return (<a onClick={this.toggleDescriptionVisibility}>{this.props.title}</a>);
```

Comment:
Should we be using anchor components here rather than `a` tags, so we can pass tracking info in if needed?